lib/OpenQA/Client/Archive.pm

```diff
@@ -165,18 +153,14 @@ sub _download_asset {
 
 }
 
-sub _download_assets {
-    my ($self, $url, $job, $path) = @_;
-
+sub _download_assets ($self, $url, $job, $path) {
     for my $asset_group (keys %{$job->{assets}}) {
         print "Attempt $asset_group download:\n";
         $self->_download_asset($url, $job->{id}, $path, $asset_group, $_) for @{$job->{assets}{$asset_group}};
     }
 }
 
-sub _progress_monitior {
-    my ($ua, $tx) = @_;
-
+sub _progress_monitior ($ua, $tx) {
```

Comment:
Just wanted to add the same note as in the previous PR - I'd add a `@` at the end for a third-party callback.
It's different to third-party methods that are called by us, or methods that we provide and are used by our customers and in our tests.
While we can always add arguments to functions optionally, a third-party module adding parameters to a callback cannot fix our wrong signature, or ignore it. (Introspection could solve that, but AFAIK we can't introspect signatures in perl yet).
In this case it might never become a problem, and if, @kraih will warn us early enough ;-)

Reply:
This is indeed a more tricky one. Not sure where i stand on the topic yet tbh. Because i doubt appending an `@` will become a best practice, since it's too situational and people will just forget it most of the time anyway. To me it seems more likely that we will end up in a future where we will just never be able to change callback arguments and have to add a whole new event.

Reply:
I don't think adding a `@` is a good idea, at least if we're going by the original intent of this ticket and its corresponding epic (https://progress.opensuse.org/issues/99579).
But I agree that changing all this any time a 3rd-party module adds some more parameters is a big hassle too. Can we somehow check for correct parameters while at the same time not breaking our code any time an outside module gets updated?